backend/src/ledgers.ts

```diff
@@ -5,37 +5,102 @@ import { Response, NextFunction } from "express";
 
 import { redisClient, getOrThrow } from "./redisSetup";
 
-import { get30DayOldLedgerQuery, bqClient, BQHistoryLedger } from "./bigQuery";
+import { getBqQueryByDate, bqClient, BQHistoryLedger } from "./bigQuery";
 
-const LEDGER_DAY_COUNT = 30;
+export enum INTERVALS {
+  hour = "hour",
+  day = "day",
+  month = "month",
+}
+
+const LEDGER_ITEM_LIMIT = {
+  hour: 60,
+  day: 24,
+  month: 30,
+};
+
+const REDIS_LEDGER_KEYS = {
+  hour: "ledgers_hour",
+  day: "ledgers_day",
+  month: "ledgers",
+};
+
+const REDIS_PAGING_TOKEN_KEY = {
+  hour: "paging_token_hour",
+  day: "paging_token_day",
+  month: "paging_token_month",
+};
+
+const HORIZON_LIMITS = {
```

Comment:
is there a reason we need different horizon limits for the different intervals? We can't use `200` for each?

Reply:
Oh, for time intervals. Since we always get the last result from each query, 200 is too much for an hour or a day and determining the time interval in a batch of 200 would be more complex than seems worth it. Considering that if we're only getting the last hour, an interval of 10 isn't much of a downside, it's still going to be really fast to get through it. Same for day.